### PR TITLE
feat(chat): show errors if there are issues when getting databases & collections to pick VSCODE-610

### PR DESCRIPTION
**Pull request by gagik**
Adds better handling of 0 collections/databases and errors to https://github.com/mongodb-js/vscode/pull/863. Instead of silently quitting, it shows the error.

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/participant/participant.ts [hunk 1]
```diff
@@ -853,26 +853,37 @@ export default class ParticipantController {
...
-    if (collections !== undefined) {
-      if (collections.length === 1) {
-        return collections[0].name;
-      }
-
+    if (collections === undefined) {
       stream.markdown(
-        `Which collection would you like to use within ${databaseName}?\n\n`
+        vscode.l10n.t(
+          `An error occurred when getting the collections from the database ${databaseName}.`
```
Comment (gagik) [inline src/participant/participant.ts:859]: do we want to send telemetrics for when this occurs?

File: src/test/suite/participant/participant.test.ts [hunk 2]
```diff
@@ -1321,6 +1321,49 @@ suite('Participant Controller Test Suite', function () {
               sinon.restore();
             });
 
+            test('shows an error if something goes wrong with getting databases', async function () {
```
Comment (gagik) [inline src/test/suite/participant/participant.test.ts:1324]: there is a lot of copy-pasting here with these tests and little change other than the command string, I could turn this into a test matrix but it'd break the structure pattern of tests.

File: src/participant/participant.ts [hunk 3]
```diff
@@ -853,26 +853,37 @@ export default class ParticipantController {
...
+      );
+      return;
+    }
+    if (collections.length === 1) {
+      return collections[0].name;
     }
 
-    return;
+    stream.markdown(
+      `Which collection would you like to use within ${databaseName}?\n\n`
```
Comment (nirinchev) [inline src/participant/participant.ts:877]: [nit] Should this be wrapped in `vscode.l10n.t`?

File: src/participant/participant.ts [hunk 4]
```diff
@@ -890,8 +901,14 @@ export default class ParticipantController {
   }): Promise<string | undefined> {
     const databases = await this._getDatabases({ stream });
 
-    if (databases === undefined || databases.length === 0) {
-      log.error('No databases found');
+    if (databases === undefined) {
+      stream.markdown(
+        vscode.l10n.t('An error occurred when getting the databases.')
```
Comment (Anemy) [inline src/participant/participant.ts:906]: Can we surface the error to users here? Something like permission issues is more transparent then.

Comment (gagik) [reply src/participant/participant.ts:906]: good point, not sure how to word it right. could something like `"An error occurred when getting the databases. You may lack the necessary permissions.` work or we can be even more explicit?

**Review by Anemy — APPROVED** [review]
lgtm! Nice error handling ux improvements

**Pull request merged**